### PR TITLE
Feature/topbar

### PR DESCRIPTION
**Pull request by Neeze**
# Description 

Create a top bar component.

***Default***

Screenshot 2023-08-25 161021

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] Already rebased against main branch.

# Screenshots

Provide screenshots or videos of the changes made if any.

### CODE REVIEW

File: packages/core-components/src/components/TopBar.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  className?: string;
+  brandIcon?: ReactNode;
+  phoneIcon?: ReactNode;
+  languageButton?: ReactNode;
+  linkContact1?: string;
+  linkContact2?: string;
+  phoneNumber1?: string;
+  phoneNumber2?: string;
+  nameContact1?: string;
+  nameContact2?: string;
```
Comment (phongduong-dp) [inline packages/core-components/src/components/TopBar.tsx:14]: @Neeze change TopBarProps only contains topBarItem: TopBarItemType

Something like this....
```suggestion
  export const TopBarItemType = {
    phoneNumber: string;
    name: string;
  }

  topBarItems[]: TopBarItemType[];
```
We do not need brandIcon by now, we need to confirm the mobile layout before implementing

File: packages/core-components/src/components/TopBar.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  phoneNumber1,
+  phoneNumber2,
+  nameContact1,
+  nameContact2,
+}) => {
+  return (
+    <div
+      className={`${className} container sm:bg-white md:bg-primary flex flex-row items-center w-screen`}
+    >
+      {mobileMode && (
```
Comment (phongduong-dp) [inline packages/core-components/src/components/TopBar.tsx:34]: @Neeze delete mobile code, we usually use tailwind to implement responsive layout, if it is not possible, we will need to create another component

File: packages/core-components/src/components/TopBar.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      )}
+      {!mobileMode && (
+        <div className={'ContactIcon basis-8 shrink-0 ' + 'sm:ms-12 sm:ml-2 '}>
+          {phoneIcon}
+        </div>
+      )}
+
+      <div
+        className={
+          'ContactInfo-1 font-Roboto text-slate-200 border-e-2 font-medium ' +
```
Comment (phongduong-dp) [inline packages/core-components/src/components/TopBar.tsx:47]: @Neeze why do we have ContactInfo-1 classname?

File: packages/core-components/src/components/TopBar.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          'hover:drop-shadow-2xl '
+        }
+      >
+        <a href={linkContact2}>
+          <span className={'sm:hidden md:contents '}>{phoneNumber2} - </span>
+          {nameContact2}
+        </a>
+      </div>
+
+      <div className={'LanguageButton scale-[95%] py-3  ' + 'ml-auto md:me-8'}>
```
Comment (phongduong-dp) [inline packages/core-components/src/components/TopBar.tsx:75]: @Neeze why do we need to concatenate string here ?

File: playground/src/topbar/TopBar.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import type { Story, StoryDefault } from "@ladle/react";
+import "regenerator-runtime/runtime";
+import "@namviet-fe/core-ui/dist/style.css";
+import { BrandIcon } from "@namviet-fe/core-ui";
+import { TopBar, type TopBarProps } from "@namviet-fe/core-ui";
+import { ContactIcon } from "@namviet-fe/core-ui";
+import { Button, type ButtonProps } from "@namviet-fe/core-ui";
+import { DropdownButton, type DropdownButtonProps } from "@namviet-fe/core-ui";
+import { FaFlagUsa } from "react-icons/fa";
```
Comment (phongduong-dp) [inline playground/src/topbar/TopBar.stories.tsx:10]: @Neeze clean up import, delete unused component and type

File: packages/core-components/src/components/TopBar.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  phoneIcon: React.ReactNode;
+  languageButton: React.ReactNode;
+}
+
+const TopBar: React.FC<TopBarProps> = ({
+  topBarItems,
+  phoneIcon,
+  languageButton,
+}) => {
+  const itemsProps = topBarItems[0];
```
Comment (its-rav) [inline packages/core-components/src/components/TopBar.tsx:25]: why do we have this topBarItems as a list only to get 1 single item ❓ 
the prop should either be a list with only 1 link/phone/name contact as its item, or a single `TopBarItemType`

File: packages/core-components/src/components/TopBar.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  return (
+    <div
+      className={`${itemsProps.className} container sm:bg-white md:bg-primary flex flex-row items-center w-screen`}
```
Comment (its-rav) [inline packages/core-components/src/components/TopBar.tsx:28]: why do the container has its bg white when at `sm` then primary color as `md` ? where do we have this?

File: packages/core-components/src/components/TopBar.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    >
+      <div className={'ContactIcon basis-8 shrink-0 ' + 'sm:ms-12 sm:ml-2 '}>
```
Comment (its-rav) [inline packages/core-components/src/components/TopBar.tsx:30]: why do we need `{}` and `+` here?

File: packages/core-components/src/components/TopBar.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        {phoneIcon}
+      </div>
+
+      <div
+        className={
+          'font-Roboto text-slate-200 border-e-2 font-medium ' +
```
Comment (its-rav) [inline packages/core-components/src/components/TopBar.tsx:36]: text color and size should be added at theme level => ask a Phong how to

File: packages/core-components/src/components/TopBar.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+          'sm:invisible md:visible ' +
```
Comment (its-rav) [inline packages/core-components/src/components/TopBar.tsx:37]: do we have this component at small screen ?

File: packages/core-components/src/components/TopBar.tsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+          'md:px-4 md:text-sm lg:text-lg ' +
+          'hover:text-white ' +
+          'hover:drop-shadow-2xl '
+        }
+      >
+        <a href={itemsProps.linkContact1}>
+          <span className={'sm:hidden md:contents'}>
+            {itemsProps.phoneNumber1} -{' '}
```
Comment (its-rav) [inline packages/core-components/src/components/TopBar.tsx:45]: why do we need this extra space ?! Can we make it work without this extra space?

File: packages/core-components/src/icons/brand.svg
Comment (its-rav) [inline packages/core-components/src/icons/brand.svg:1]: I dont see we use this brand.svg here anywhere!?

File: playground/src/topbar/TopBar.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import type { Story, StoryDefault } from "@ladle/react";
+import "regenerator-runtime/runtime";
+import "@namviet-fe/core-ui/dist/style.css";
+import { type TopBarProps, TopBar } from "@namviet-fe/core-ui";
+import { ContactIcon } from "@namviet-fe/core-ui";
+import { DropdownButton, type DropdownButtonProps } from "@namviet-fe/core-ui";
+import { EnIcon, ViIcon } from "@namviet-fe/core-ui";
+import { LanguageSelect } from "../buttons/Select.stories";
```
Comment (its-rav) [inline playground/src/topbar/TopBar.stories.tsx:9]: unuse import

**Review by its-rav — CHANGES_REQUESTED** [review]

File: packages/core-components/src/components/TopBar.tsx (hunk 10 above)
Comment (Neeze) [reply packages/core-components/src/components/TopBar.tsx:37]: I heard that maybe we have to create new component top bar for mobile so I think we don't need this component at small screen size

File: packages/core-components/src/components/TopBar.tsx (hunk 11 above)
Comment (Neeze) [reply packages/core-components/src/components/TopBar.tsx:45]: I'll fix that

**Review by Neeze — COMMENTED** [review]
I almost fixed what you commented but "text color and size" I will ask @phongduong-dp to solve it soon, @its-rav  you can re-review it.

**Comment by phongduong-dp** [comment]
@Neeze Can you merge develop branch and resolve any conflict that happens !

File: packages/core-components/src/components/TopBar.tsx [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    <div
+      className={`${className} container sm:bg-primary flex flex-row items-center w-screen`}
+    >
+      <div
+        className={'basis-8 shrink-0 sm:invisible md:visible md:ms-12 md:ml-2 '}
+      >
+        {phoneIcon}
+      </div>
+
+      <div className={'flex justify-center items-center'}>
```
[the remark this answers is not in the document]
Comment (Neeze) [reply packages/core-components/src/components/TopBar.tsx:32]: Do you mean the string in className or which line?

File: playground/src/topbar/TopBar.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import type { Story, StoryDefault } from "@ladle/react";
+import "regenerator-runtime/runtime";
+import "@namviet-fe/core-ui/dist/style.css";
+import { type TopBarProps, TopBar } from "@namviet-fe/core-ui";
```
[the remark this answers is not in the document]
Comment (Neeze) [reply playground/src/topbar/TopBar.stories.tsx:5]: I solved it

File: packages/core-components/src/components/TopBar.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (phongduong-dp) [reply packages/core-components/src/components/TopBar.tsx:31]: > Do you mean the string in className or which line?
String in className, if we need to write TSX code then {} is required, if not, then simple string is enough, remove the {}

**Comment by its-rav** [comment]
inactivity

**Pull request closed without merging**